src/test/java/fr/inria/gforge/spoon/mojo/SpoonMojoTest.java

```diff
@@ -252,4 +253,39 @@ public void testSpoonGoalGenerateResultFileForProjectWithGeneratedSources() thro
 		assertThat(files.length).isEqualTo(1);
 		assertThat(files[0].getName()).startsWith("result-spoon");
 	}
+
+	@Test
+	public void testSpoonGoalWithTestClass() throws Exception {
+		File basedir = resources.getBasedir("hello-world-with-test");
+		rule.executeMojo(basedir, "generate");
+
+		final File dirOutputResults = new File(basedir, "target/spoon-maven-plugin");
+		assertThat(dirOutputResults).exists();
+
+		final File[] files = dirOutputResults.listFiles();
+		assertThat(files.length).isEqualTo(1);
+		assertThat(files[0].getName()).startsWith("result-spoon");
+
+		final File contentSource = new File(basedir, "target/generated-sources/spoon/fr/inria/gforge/spoon");
+		assertThat(contentSource).exists();
+
+		final File[] sourceFiles = contentSource.listFiles();
+		assertThat(sourceFiles.length).isEqualTo(2);
+		assertTrue(Arrays.stream(sourceFiles).allMatch(v->v.getName().equals("AppTest.java") || v.getName().equals("App.java")));
+	}
+
+	@Test
+	public void testSpoonCheckGoalWithTest() throws Exception {
+		File basedir = resources.getBasedir("hello-world-with-test");
+		rule.executeMojo(basedir, "check");
+
+		final File dirOutputResults = new File(basedir, "target/spoon-maven-plugin");
+		assertThat(dirOutputResults).exists();
+
+		final File contentSource = new File(basedir, "target/generated-sources/spoon/fr/inria/gforge/spoon");
+		assertThat(contentSource).doesNotExist();
+		
+		final File resultFileProcessor = new File(basedir, "target/spoon-maven-plugin/spoon-nb-statement.txt");
+		assertThat(resultFileProcessor).exists();
```

Comment:
I think this test will always pass. Because you have an App.java class and thus the resultFileProcessor will always exist. Am I wrong?

Reply:
from my understanding the test will always fail, because we need the processors from the "/resources/processors/"

Reply:
Note: on my side I didn't try the test I wrote because the maven build was failing.

The processor is at https://github.com/SpoonLabs/spoon-maven-plugin/blob/4bd92ae585a0004ec3c587445c71c5dd3f7c5133/src/test/java/fr/inria/gforge/spoon/mojo/CountStatementProcessor.java#L10, ,no? So it should be there in the CP.

What I meant is that you're testing that the check worked by verifying that the output file exists. But this file will be created by the processor running on the main source, no?

Side note: this is why I didn't have any App.java class in my test.

Reply:
All this is speculation since I didn't run the maven build :)

Reply:
having only test classes is a way better test, good point. Changed it.